sql/src/main/java/io/crate/planner/consumer/UpdatePlanner.java

```diff
@@ -184,19 +200,48 @@ private static ExecutionPlan sysUpdate(PlannerContext plannerContext,
                                            Map<Reference, Symbol> assignmentByTargetCol,
                                            Symbol query,
                                            Row params,
-                                           SubQueryResults subQueryResults) {
+                                           SubQueryResults subQueryResults,
+                                           @Nullable List<Symbol> returnValues,
+                                           @Nullable List<Field> outputFields) {
         TableInfo tableInfo = table.tableInfo();
         Reference idReference = requireNonNull(tableInfo.getReference(DocSysColumns.ID), "Table must have a _id column");
-        SysUpdateProjection updateProjection = new SysUpdateProjection(idReference.valueType(), assignmentByTargetCol);
+        Symbol[] outputSymbols;
+        if (returnValues == null) {
+            outputSymbols = new Symbol[]{new InputColumn(0, DataTypes.LONG)};
+        } else {
+            outputSymbols = new Symbol[outputFields.size()];
+            for (int i = 0; i < outputFields.size(); i++) {
+                outputSymbols[i] = new InputColumn(i, outputFields.get(i).valueType());
+            }
+        }
+        SysUpdateProjection updateProjection = new SysUpdateProjection(new InputColumn(0, idReference.valueType()),
```

Comment:
Style nitpick and maybe something we should add a checkstyle rule for or change intellij defaults:

Could put the arguments into a separate line to reduce the level of indentation, so that instead of

```
#############################################################################
                                                      #######################
                                                      #######################
                                                      #######################
                                                      #######################
                                                      #######################
                                                      #######################
                                                      #######################
                                                      #######################
```

We've

```
#####################################################
    #####################
    #####################
    #####################
    ...
```


Most people tend to read from left to right. Having the arguments further to
the left helps with skimming.

Reply:
I fully agree on this.